# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/TlsHandshaker.cpp
# Commit: 00bb17e2a6a7
# Full Hash: 00bb17e2a6a73f9c66e861adae426818ff4d8caa
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2023-09-21 21:20:56
# Regressor Bug: 1784261
# File Overlap Count: 1
# Description:
#   Bug 1784261 - Add telemetry to understand the reason of connection closure, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D188132
# ==============================================================================

diff -r e98350675c5a -r 00bb17e2a6a7 netwerk/protocol/http/TlsHandshaker.cpp
--- a/netwerk/protocol/http/TlsHandshaker.cpp	Thu Sep 21 08:42:26 2023 +0000
+++ b/netwerk/protocol/http/TlsHandshaker.cpp	Thu Sep 21 08:59:03 2023 +0000
@@ -258,6 +258,7 @@
          "early selected alpn not available",
          mOwner.get()));
   } else {
+    mOwner->ChangeConnectionState(ConnectionState::ZERORTT);
     LOG1(
         ("TlsHandshaker::Check0RttEnabled %p -"
          "early selected alpn: %s",